### PR TITLE
Ft- passwordReset

### PR DESCRIPTION
**Pull request by Hurntre**
added the password reset feature

#### Type of change

- [ ]
- [ ]

#### How Has This Been Tested?

- [ ] Unit testing

#### PT-ID

[](url)

#### Screenshots
image
image
image
image
image

#### Questions:

### CODE REVIEW

File: server/controllers/auth.js [hunk 1]
```diff
@@ -54,9 +57,107 @@ const loginController = async (req, res) => {
...
+    });
+    if (!user) {
+      return res.status(400).send({
+        success: false,
+        error: `Email not found`,
+      });
+    }
+    const { _id } = user;
+    const token = tokenGenerator.userToken({ _id, email });
+    const link = `http://localhost:3000/api/v1/auth/reset/${token}`;
```
Comment (vic3king) [inline server/controllers/auth.js]: Please add this URL to your env, e.g FRONTEND_URL || RESET_URL = http://localhost:3000/api/v1/auth/reset/, import it here and add the token e.g `RESET_URL${token}`

File: server/helpers/emailService.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import nodeMailer from 'nodemailer';
+
+const env = process.env.NODE_ENV;
+const resetMailOptions = (email, firstName, link) => {
```
Comment (vic3king) [inline server/helpers/emailService.js]: Please move this function to a separate file probably called notifications, this file will hold subsequent notification functions, like signup etc

File: tests/integration/reset-password.test.js [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+        expect(message).to.equal(
+          'A reset email has been sent to email56@yahoo.com.'
+        );
+        expect(data).contains({
+          message,
+        });
+        done();
+      });
+  });
+  // it('should throw an error when there is no internet connection', done => {
```
Comment (vic3king) [inline tests/integration/reset-password.test.js]: You can delete this test block if it's no longer needed.

**Comment by Hurntre** [comment]
The reviews have been resolved

**Pull request merged**